### PR TITLE
Instantiate GrowingVectorMemory for CUDAWrappers::Vector

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: source/lac/vector_memory.inst.in [hunk 1]
```diff
@@ -20,3 +20,9 @@ for (VECTOR : VECTOR_TYPES)
     template class VectorMemory<VECTOR>;
     template class GrowingVectorMemory<VECTOR>;
 }
+
+for (SCALAR : REAL_SCALARS)
+{
+    template class VectorMemory<LinearAlgebra::CUDAWrappers::Vector<SCALAR>>;
```
Comment (tjhei) [inline source/lac/vector_memory.inst.in]: For other types we only instantiate things if we have the external dependency. Would it make sense to guard this with ``DEAL_II_WITH_CUDA``?

Comment (Rombur) [reply source/lac/vector_memory.inst.in]: yeah good catch

File: source/lac/vector_memory.inst.in [hunk 2]
```diff
@@ -20,3 +20,13 @@ for (VECTOR : VECTOR_TYPES)
     template class VectorMemory<VECTOR>;
     template class GrowingVectorMemory<VECTOR>;
 }
+
+#ifdef DEAL_II_WITH_MPI
```
Comment (bangerth) [inline source/lac/vector_memory.inst.in]: I think this should be `DEAL_II_WITH_CUDA`, no?

**Comment by masterleinad** [comment]
@bangerth Would you mind to have another look?

**Pull request merged**